src/main/java/de/zalando/aruha/nakadi/service/subscription/state/StartingState.java

```diff
@@ -0,0 +1,53 @@
+package de.zalando.aruha.nakadi.service.subscription.state;
+
+import de.zalando.aruha.nakadi.exceptions.NoStreamingSlotsAvailable;
+import de.zalando.aruha.nakadi.service.subscription.model.Partition;
+import de.zalando.aruha.nakadi.service.subscription.model.Session;
+import java.io.IOException;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public class StartingState extends State {
+    private static final Logger LOG = LoggerFactory.getLogger(StartingState.class);
+
+    @Override
+    public void onEnter() {
+        getZk().runLocked(this::createSubscriptionLocked);
+    }
+
+    /**
+     * 1. Checks, that subscription node is present in zk. If not - creates it.
+     * <p>
+     * 2. Registers session.
+     * <p>
+     * 3. Switches to streaming state.
+     */
+    private void createSubscriptionLocked() {
+        // check that subscription initialized in zk.
+        if (getZk().createSubscription()) {
+            // if not - create subscription node etc.
+            getZk().fillEmptySubscription(getKafka().getSubscriptionOffsets());
+        } else {
+            final Session[] sessions = getZk().listSessions();
+            final Partition[] partitions = getZk().listPartitions();
+            if (sessions == null || partitions == null) {
```

Comment:
I think you have some unnecessary null checks (not only here). I don't see any cases where that can be null.
Moreover in Nakadi code we have a following null-policy:
If some method can return null - it should be marked as @Nullable. Everything else that is not marked @Nullable is considered nonnull by default and does not require null checks.